pkg/controller/jobframework/validation.go

```diff
@@ -144,3 +145,28 @@ func validateUpdateForMaxExecTime(oldJob, newJob GenericJob) field.ErrorList {
 	}
 	return nil
 }
+
+// ValidateImmutablePodSpec function is used for serving workloads to ensure no changes are allowed
+// to the PodSpec except for the image field in containers.
+func ValidateImmutablePodSpec(newPodSpec *corev1.PodSpec, oldPodSpec *corev1.PodSpec, fieldPath *field.Path) field.ErrorList {
```

Comment:
Is this function covered with unit tests? It could be either in validation_test.go or in lws webhook tests, but some tests will be helpful.

Reply:
I see, I think the ones are enough for a starting point.